src/main/java/core/basesyntax/Pair.java

```diff
@@ -1,5 +1,49 @@
 package core.basesyntax;
 
-class Pair {
+class Pair<K, V> {
+    private K first;
+    private V second;
 
+    public Pair(K first, V second) {
+        this.first = first;
+        this.second = second;
+    }
+
+    K getFirst() {
+        return first;
+    }
+
+    V getSecond() {
+        return second;
+    }
+
+    static <K, V> Pair of(K first, V second) {
+        return new Pair(first, second);
+    }
+
+    @Override
+    public boolean equals(Object object) {
+        if (this == object) {
+            return true;
+        }
+
+        if (object == null || getClass() != object.getClass()) {
+            return false;
+        }
+
+        if (object.getClass().equals(Pair.class)) {
+            Pair pair = (Pair) object;
+            return first == null ? pair.first == null : first.equals(pair.first)
```

Comment:
ry to make this check without ternary operator, just use logical operators && and ||